src/Xamarin.Android.Tools.Bytecode/ClassPath.cs

```diff
@@ -46,7 +46,17 @@ public void Load (string jarFile)
 			if (!IsJarFile (jarFile))
 				throw new ArgumentException ("'jarFile' is not a valid .jar file.", "jarFile");
 
-			using (var jar = CreateZipArchive (jarFile)) {
+			using (var jarStream = File.OpenRead (jarFile)) { 
+				Load (jarStream);
+			}
+		}
+
+		public void Load (Stream jarStream)
```

Comment:
I believe this should follow the [`StreamWriter(Stream, Encoding, Int32, Boolean)`](https://docs.microsoft.com/en-us/dotnet/api/system.io.streamwriter.-ctor?view=netframework-4.7.2#System_IO_StreamWriter__ctor_System_IO_Stream_System_Text_Encoding_System_Int32_System_Boolean_) constructor and have a public `leaveOpen` parameter, and forward this through:

```csharp
public void Load (Stream jarStream, bool leaveOpen = false)
```